requirements/ci-requirements.txt

```diff
@@ -1,10 +1,10 @@
 
 # Linters and such. Pin them so that different devs
 # don't get different results from using them.
-flake8==5.0.4
-mypy==0.982
-black==22.6.0
-isort==5.10.1
+flake8==7.1.1
+mypy==1.13.0
+black==24.10.0
+isort==5.13.2
```

Comment:
Still needed after introducing ruff?

Reply:
I'll remove them once I get stuff working with the uv toolchain in CI; coming in a separate PR